docs: add contributing including release process (#571)
aneoconsulting/ArmoniK.Core: CONTRIBUTING.md

# Contributing Guide

> Hi! We are very happy that you are interested in contributing to ArmoniK.Api. Before submitting your contribution, please make sure to take a moment and read through the following guide.

Reviewer: ```suggestion
Hi! We are very happy that you are interested in contributing to ArmoniK.Core project. Before submitting your contribution, please make sure to take a moment and read through the following guide.
```

## Pull Request Process

1. Ensure any install or build dependencies are removed before the end of the layer when doing a
   build.
2. Update the README.md with details of changes to the interface, this includes new environment
   variables, exposed ports, useful file locations and container parameters.
3. Make sure to do a PR against main
4. Make sure you clearly define the changes in the description of the PR
5. Make sure all tests pass

## Release Process

When necessary, maintainers can release a new version. This new version will publish packages to registries.

### Release a new version

> Replace <version> with the new version number

> 1. Install .Net latest release version (https://dotnet.microsoft.com/en-us/download)

Reviewer: not sure that we should always target the latest release, eventually the latest stable.

Author: that is for the compiler. The target is defined by the project and you cannot change it without modifying code

> 2. Make sure that ArmoniK.Api.* versions are releases

Reviewer: ```suggestion
2. Make sure that ArmoniK.Core.* versions are releases
```

Author: core depends on releases from API

>   1. `git grep '"ArmoniK\.Api\.'` to check which version is used. You can also see that with any editor in csproj files.

Reviewer: ```suggestion
  1. `git grep '"ArmoniK\.Core\.'` to check which version is used. You can also see that with any editor in csproj files.
```

Author: that does not make any sense here

Reviewer: my bad
